spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/connection/AxonServerDockerComposeConnectionDetailsFactory.java

```diff
@@ -0,0 +1,39 @@
+package org.axonframework.springboot.connection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.docker.compose.service.connection.DockerComposeConnectionDetailsFactory;
+import org.springframework.boot.docker.compose.service.connection.DockerComposeConnectionSource;
+
+/**
+ * ConnectionDetailsFactory implementation that recognizes Spring Boot Docker Compose-managed Docker containers running
+ * Axon Server and creates an AxonServerConnectionDetails bean with the details to connect to AxonServer within that
+ * container.
+ *
+ * @author Allard Buijze
+ * @since 4.9
+ */
+public class AxonServerDockerComposeConnectionDetailsFactory extends DockerComposeConnectionDetailsFactory<AxonServerConnectionDetails> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AxonServerDockerComposeConnectionDetailsFactory.class);
+
+    /**
+     * Initializes the factory to look for containers running the "axoniq/axonserver" image.
+     */
+    public AxonServerDockerComposeConnectionDetailsFactory() {
+        super("axoniq/axonserver", "io.axoniq.axonserver.connector.AxonServerConnectionFactory");
+    }
+
+    @Override
+    protected AxonServerConnectionDetails getDockerComposeConnectionDetails(DockerComposeConnectionSource source) {
+        int port = source.getRunningService().ports().get(8124);
+        int uiPort = source.getRunningService().ports().get(8024);
```

Comment:
Curious: what happens when somebody changes the ports to something different than 8124 and 8024?

Reply:
Then the `get` method throws an IllegalArgumentException. But the only people that could change this, are the creators of the Axon Server container.